Detectors/ITSMFT/common/reconstruction/include/ITSMFTReconstruction/ChipMappingMFT.h

```diff
@@ -25,113 +25,251 @@ namespace itsmft
+  {
+    return ruInfo.firstChipIDSW + mCableHWFirstChip[ruInfo.ruType][cableHW] + chipModuleIDHW2SW(ruInfo.ruType, chOnModuleHW);
+  }
+
+  ///< convert HW id of chip in the module to SW ID (sequential ID on the module)
+  int chipModuleIDHW2SW(int ruType, int hwIDinMod) const
   {
-    return 0;
+    return hwIDinMod;
   }
 
-  static constexpr int getNChips() { return NChips; }
+  ///< convert SW id of chip in the module to HW ID
+  int chipModuleIDSW2HW(int ruType, int swIDinMod) const
+  {
+    return swIDinMod;
+  }
 
-  // RS placeholder for methods to implement -----------<
+  static constexpr Int_t getNChips() { return NChips; }
 
-  static constexpr int getNModules() { return NModules; }
+  static constexpr Int_t getNModules() { return NModules; }
 
-  int chipID2Module(int chipID, int& chipInModule) const
+  Int_t chipID2Module(Int_t chipID, Int_t& chipOnModule) const
   {
-    chipInModule = ChipMappingData[chipID].chipInModule;
+    chipOnModule = ChipMappingData[chipID].chipOnModule;
     return ChipMappingData[chipID].module;
   }
 
-  int chipID2Module(int chipID) const
+  Int_t chipID2Module(Int_t chipID) const
   {
     return ChipMappingData[chipID].module;
   }
 
-  int getCablesOnRUType(int ruType) const { return 0; }
-
-  int getNChipsInModule(int modID) const
+  Int_t getNChipsInModule(Int_t modID) const
   {
     return ModuleMappingData[modID].nChips;
   }
 
-  int module2ChipID(int modID, int chipInModule) const
+  Int_t module2ChipID(Int_t modID, Int_t chipOnModule) const
   {
-    return ModuleMappingData[modID].firstChipID + chipInModule;
+    return ModuleMappingData[modID].firstChipID + chipOnModule;
   }
 
-  int module2Layer(int modID) const
+  Int_t module2Layer(Int_t modID) const
   {
     return ModuleMappingData[modID].layer;
   }
 
-  int chip2Layer(int chipID) const
+  Int_t chip2Layer(Int_t chipID) const
   {
     return ModuleMappingData[ChipMappingData[chipID].module].layer;
   }
 
   ///< impose user defined FEEId -> ruSW (staveID) conversion, to be used only for forced decoding of corrupted data
   void imposeFEEId2RUSW(uint16_t, uint16_t) {}
 
+  ///< extract information about the chip with SW ID
+  void getChipInfoSW(Int_t chipSW, ChipInfo& chInfo) const
+  {
+    UShort_t ladder = ChipMappingData[chipSW].module;
+    UChar_t layer = ModuleMappingData[ladder].layer;
+    UChar_t zone = ModuleMappingData[ladder].zone;
+    UChar_t half = ModuleMappingData[ladder].half;
+
+    chInfo.ruType = ZoneRUType[zone][layer / 2];
+
+    // count RU SW per half layers
+    //chInfo.ru = NLayers * (NZonesPerLayer / 2) * half + (NZonesPerLayer / 2) * layer + zone;
+
+    // count RU SW per full layers
+    chInfo.ru = NZonesPerLayer * layer + (NZonesPerLayer / 2) * half + zone;
+
+    chInfo.id = ChipMappingData[chipSW].chipOnRU;
+    chInfo.chOnRU = getChipOnRUInfo(chInfo.ruType, chInfo.id);
+    printf("BV===== getChipInfoSW for chipID %3d RU type %2d chip on RU SW %2d \n", chipSW, chInfo.ruType, chInfo.id);
```

Comment:
do you need this printout permanently. Consider using LOG(DEBUG) or similar.

Reply:
Sorry, I thought I have cleaned everything.